OpenRA.Mods.Common/Activities/PickupUnit.cs

```diff
@@ -74,26 +76,31 @@ public override bool Tick(Actor self)
 				return true;
 			}
 
+			// Abort if the cargo has canceled the reservation
```

Comment:
It seems this entire branch can be removed, because the `cargo != carryall.Carryable` in L62 should already catch all the cases that this branch would catch. When the carryable is unreserved, cargo is set to null.

Reply:
Done.